cli/src/cluster_query.rs

```diff
@@ -1170,6 +1219,37 @@ pub fn process_show_validators(
     Ok("".to_string())
 }
 
+pub fn process_transaction_history(
+    rpc_client: &RpcClient,
+    address: &Pubkey,
+    start_slot: Option<Slot>,
+    end_slot: Option<Slot>,
+) -> ProcessResult {
+    let (start_slot, end_slot) = {
+        if let Some(start_slot) = start_slot {
+            if let Some(end_slot) = end_slot {
+                (start_slot, end_slot)
+            } else {
+                (start_slot, start_slot + 1000)
+            }
+        } else {
+            let current_slot = rpc_client.get_slot_with_commitment(CommitmentConfig::max())?;
+            (current_slot.saturating_sub(1000), current_slot)
+        }
+    };
+
+    println!(
+        "Transactions affecting {} within slots [{},{}]",
+        address, start_slot, end_slot
+    );
```

Comment:
This is one subcommand for which it could be useful to return json. Could we create a Cli type and impl Display for niceties like this print statement and the signatures count below? (a la #9478 )

Reply:
I thought about that but didn't go for it since this command really is just cli window dressing over the same JSON RPC API.    But my mind is open to be convinced

Reply:
Okay, fair. I think since this has the potential to return a whole heap of data, request for json might come up, but we can refactor then.